### DISCUSSION TITLE
Seed Scalar and Seed Input for Samplers

### DISCUSSION THREAD

**Discussion by WASasquatch** [comfyanonymous/ComfyUI#102] [Ideas]
A Seed Scalar output node for seeds, and input node for samplers. That way, when doing a HR fix type img2img after first generation, you can preserve the seed easily between both samplers. Bonus if Seed Scalar had a randomize button, but not on each gen (otherwise it would change before img2img)

**Comment by fambaa** [comment]
An optional input for samplers for a node is a great idea. It should also have the randomize option, but when you connect the seed node to multiple samplers it passes on the same seed to all.

**Reply by WASasquatch** [reply]
Yeah that would be perfect and optimal.

**Comment by WASasquatch** [comment]
Optional input is not implemented yet, but I have made a external seed based Sampler you can download here: https://civitai.com/models/20793/was-node-suites-comfyui
